lib/modules/manager/maven/lockfile.ts

```diff
@@ -0,0 +1,92 @@
+import util from 'util';
+import glob from 'glob';
+import type { StatusResult } from 'simple-git';
+import { logger } from '../../../logger';
+import { exec } from '../../../util/exec';
+import type { ExecResult } from '../../../util/exec/types';
+import { readLocalFile } from '../../../util/fs';
+import { getRepoStatus } from '../../../util/git';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+const maven_lockfile_version = '4.0.0';
```

Comment:
shouldn't this be extracted from package file?

Reply:
We currently don't include the version of the lockfile generator in the lockfile. But given your comment, this seems needed.

Reply:
I have updated this to use a maven command resolving the declared version of the plugin.